api/src/main/java/me/lokka30/treasury/api/economy/account/accessor/NonPlayerAccountAccessor.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * This file is/was part of Treasury. To read more information about Treasury such as its licensing, see <https://github.com/lokka30/Treasury>.
+ */
+
+package me.lokka30.treasury.api.economy.account.accessor;
+
+import java.util.Objects;
+import java.util.concurrent.CompletableFuture;
+import me.lokka30.treasury.api.common.response.Response;
+import me.lokka30.treasury.api.economy.account.NonPlayerAccount;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Represents an account accessor, which accesses {@link NonPlayerAccount non player accounts}
+ *
+ * @author MrIvanPlays
+ * @since 2.0.0
+ */
+public abstract class NonPlayerAccountAccessor {
```

Comment:
Add method `fromData(@NotNull AccountData)`, which provides an accessor for the account identified by the provided AccountData object.

Reply:
this is not meant to be instantiated by economy consumers, only by economy providers.